src/NUnitFramework/tests/Api/TestAssemblyRunnerTests.cs

```diff
@@ -95,6 +95,7 @@ public void Load_GoodFile_ReturnsRunnableSuite()
         }
 
         [Test]
+        [SetUICulture("en-US")]
```

Comment:
Once we enable the SetCulture tests, I'm really excited about this PR and I'll want to get it in.

However, I would like us to review each of these changes on a case-by-case basis with this question in mind: is the right fix to change the framework to use an ordinal comparison? For example, with files, I would say that should absolutely be an OrdinalIgnoreCase comparison, never culture-dependent.

Reply:
>For example, with files, I would say that should absolutely be an OrdinalIgnoreCase comparison, never culture-dependent.

It's not about ordinal comparison but mostly about exception messages that are localized.

This test (SetUICulture commented out):

```csharp
private const string COULD_NOT_LOAD_MSG = "Could not load";
private const string MISSING_FILE = "junk.dll";
private static readonly IDictionary<string, object> EMPTY_SETTINGS = new Dictionary<string, object>();

[Test]
//[SetUICulture("en-US")]
public void Load_FileNotFound_ReturnsNonRunnableSuite()
{
    var result = _runner.Load(MISSING_FILE, EMPTY_SETTINGS);

    Assert.That(result.IsSuite);
    Assert.That(result, Is.TypeOf<TestAssembly>());
    Assert.That(result.Name, Is.EqualTo(MISSING_FILE));
    Assert.That(result.RunState, Is.EqualTo(Interfaces.RunState.NotRunnable));
    Assert.That(result.TestCaseCount, Is.EqualTo(0));
    Assert.That(result.Properties.Get(PropertyNames.SkipReason),
        Does.StartWith(COULD_NOT_LOAD_MSG));
}
```

produces the following test failure:

```
1) Failed : NUnit.Framework.Api.TestAssemblyRunnerTests.Load_FileNotFound_ReturnsNonRunnableSuite
  Expected: String starting with "Could not load"
  But was:  "Die Datei oder Assembly "junk.dll" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden.
  ----> Das System kann die angegebene Datei nicht finden. (Ausnahme von HRESULT: 0x80070002)"
   bei NUnit.Framework.Api.TestAssemblyRunnerTests.Load_FileNotFound_ReturnsNonRunnableSuite() in C:\Users\Brar\Documents\GitHub\nunit\src\NUnitFramework\tests\Api\TestAssemblyRunnerTests.cs:Zeile 110.
```

While we could certainly assert on the non-localized elements that are contained in both strings it wouldn't be much more than the file name in this case.

With `[SetUICulture("en-US")]` (funny that it's still partially localized):
```
Could not load file or assembly 'junk.dll' or one of its dependencies. Das System kann die angegebene Datei nicht finden.
  ----> Das System kann die angegebene Datei nicht finden. (Exception from HRESULT: 0x80070002)
```

Without `[SetUICulture("en-US")]`:
```
Die Datei oder Assembly "junk.dll" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden.
  ----> Das System kann die angegebene Datei nicht finden. (Ausnahme von HRESULT: 0x80070002)
```

Reply:
BTW: Can anyone tell me why/how the test cited above (and quite a few others) aren't executed with `.\build.ps1 -t TestNetStandard14`?

I get
```
Test Count: 5064, Passed: 4939, Failed: 0, Warnings: 0, Inconclusive: 109, Skipped: 16
    Skipped Tests - Ignored: 3, Explicit: 13, Other: 0
```
for `.\build.ps1 -t Test45` and
```
  Test Count: 2719, Passed: 2699, Failed: 0, Warnings: 0, Inconclusive: 0, Skipped: 20
    Skipped Tests - Ignored: 14, Explicit: 6, Other: 0
```
for `.\build.ps1 -t TestNetStandard14`.

There are no `#if` definitions disabling the aforementioned test for a certain target and I have even confirmed (via ildasm) that it's in the compiled `nunit.framework.tests.dll` for `netcoreapp1.1`, yet it isn't executed.

Reply:
> It's not about ordinal comparison but mostly about exception messages that are localized.

Thanks for confirming. Each of these cases is indeed about localized exception messages, and this is the right fix.

> BTW: Can anyone tell me why/how the test cited above (and quite a few others) aren't executed

That's concerning; taking a look.

Reply:
The tests were not running for netstandard1.6 prior to my PR. I think I know which PR this was, though

Reply:
Confirmed. https://github.com/nunit/nunit/pull/2796 introduced the regression.